source/collection-management/tests/modifyCollection-errorResponse.yml

```diff
@@ -0,0 +1,59 @@
+description: "modifyCollection-errorResponse"
+
+schemaVersion: "1.12"
+
+createEntities:
+  - client:
+      id: &client0 client0
+      observeEvents: [ commandStartedEvent ]
+  - database:
+      id: &database0 database0
+      client: *client0
+      databaseName: &database0Name collMod-tests
+  - collection:
+      id: &collection0 collection0
+      database: *database0
+      collectionName: &collection0Name test
+
+initialData: &initialData
+  - collectionName: *collection0Name
+    databaseName: *database0Name
+    documents:
+      - { _id: 1, x: 1 }
+      - { _id: 2, x: 1 }
+
+tests:
+  - description: "modifyCollection prepareUnique violations are accessible"
```

Comment:
This test covers the original impetus for this functionality (see: [DRIVERS-2353](https://jira.mongodb.org/browse/DRIVERS-2353)). The other tests for `aggregate` and `findOneAndUpdate` are meant to provide some additional test coverage without being exhaustive.